Makefile.core.mk

```diff
@@ -143,10 +143,17 @@ export
 ##@ Testing
 
 .PHONY: test
-test: envtest ## Run unit tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" \
+test: test.unit test.integration ## Run both unit tests and integration test.
+
+.PHONY: test.unit
+test.unit:  ## Run unit tests.
 	go test $(VERBOSE_FLAG) ./... -coverprofile cover.out
 
+.PHONY: test.integration
+test.integration: envtest ## Run integration tests located in the tests/integration directory.
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" \
+	go test --tags=integration $(VERBOSE_FLAG) ./tests/integration/... -coverprofile cover.out
```

Comment:
Isn't it redundant to have go tags and invoking go test only for the ./tests/integration/ directory?
cc @luksa

Reply:
Yes, it's redundant. I can run the go test on `./...` but the output can be a little confusing because you will see the check over all the test files without executing anything in those who do not have the tag integration. if you see it Ok I can add it in another PR

Reply:
It's not redundant. The `--tags=integration` must be present for the tests to be built.

Reply:
If you see it, it's redundant because we can delete the build tag and run the same command in the same folder and the result will be the same. But having the tag is useful

Reply:
Delete the build tag from where? From the command in this make target? If you do this, no integration test will run, because it won't be compiled.